build.go

```diff
@@ -257,17 +252,16 @@ func runComposerInstall(
 		logger.Process("Reusing cached layer %s", composerPackagesLayer.Path)
 		logger.Break()
 
-		composerPackagesLayer.Launch, composerPackagesLayer.Build, composerPackagesLayer.Cache = launch, build, build
+		composerPackagesLayer.Launch, composerPackagesLayer.Build, composerPackagesLayer.Cache = launch, build, launch
```

Comment:
right, I was confused about this as well due to [this conversation](https://github.com/paketo-buildpacks/composer-install/pull/2/files/8296cc5c8b767d93446f9cd636d02073f4a5f452#r854485343). 

However, when `cache` was set to `build` (as it was before), layer reuse now fails with the following error:
```
            [builder]   Reusing cached layer /layers/paketo-buildpacks_composer-install/composer-packages
            [builder]
            [builder]   Detected existing vendored packages, replacing with cached vendored packages
            [builder] stat /layers/paketo-buildpacks_composer-install/composer-packages/vendor: no such file or directory
```

Reply:
For some reason, the way we are now handling the layers causes the launch metadata to not be restored as it was before in this explanation:

> We never need to set the cache layer to the value of launch. It's confusing but when launch=true, the layer metadata will always be restored, the buildpack reads the metadata and leaves it in place, telling the lifecycle to reuse the layer. This is all to say that we don't need to set cache to launch, because we will get this behaviour by proxy of the layer reuse code in the buildpack + how the lifecycle treats launch layers.


I'm not sure if this means we are doing something wrong or if this is expected behaviour

Reply:
@thitch97 and I have to the conclusion that we now need cache=true because we are copying from the layer directory into workspace on rebuilds now. Previously, we only needed launch=true, because we didn't touch the layer directory on rebuilds, just set up a symlink which took effect at launch-time.

Reply:
What happens in the case of a build-time only dependence on the composer packages? Like `build=true` and `launch=false`?

Reply:
@ryanmoran the diff here is outdated, and we decided to set `cache=true` in order to cover both cases. Is that a concern?